.github/workflows/pull-request.yml

```diff
@@ -69,7 +69,7 @@ jobs:
 
       - name: Archive lint results
         if: always()
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
           name: lint-report
```

Comment:
[Breaking changes in v4](https://github.com/actions/upload-artifact?tab=readme-ov-file#v4---whats-new) 

"Due to how Artifacts are created in this new version, it is no longer possible to upload to the same named Artifact multiple times. You must either split the uploads into multiple Artifacts with different names, or only upload once. Otherwise you will encounter an error." - this may cause us an issue